app/services/jobseekers/job_applications/quick_apply.rb

```diff
@@ -7,17 +7,72 @@ def initialize(jobseeker, vacancy)
   end
 
   def job_application
-    new_job_application.assign_attributes(recent_job_application.slice(*attributes_to_copy).merge(completed_steps: completed_steps, in_progress_steps: in_progress_steps))
-    copy_qualifications
-    copy_employments
-    copy_references
-    copy_training_and_cpds
+    return new_job_application unless previously_submitted_application? || jobseeker_profile
+
+    source = previously_submitted_application? ? recent_job_application : jobseeker_profile
```

Comment:
Suggestion :raising_hand_man: 
I think you could avoid the extra query triggered by `previously_submitted_application?` If you directly use `recent_job_application` (that may be `nil` if no recent applications exist).
```ruby
    source = recent_job_application || jobseeker_profile
    return new_job_application unless source
```

Then you could get rid of `previously_submitted_application?` definition.